src/testing/core/createSchemaFetch.ts

```diff
@@ -32,14 +34,31 @@ import { withCleanup } from "../internal/index.js";
  */
 const createSchemaFetch = (
   schema: GraphQLSchema,
-  mockFetchOpts: { validate: boolean } = { validate: true }
+  mockFetchOpts: {
+    validate?: boolean;
+    delay?: { min: number; max: number };
+  } = { validate: true, delay: { min: 0, max: 0 } }
 ) => {
   const prevFetch = window.fetch;
 
-  const mockFetch: (uri: any, options: any) => Promise<Response> = (
+  const mockFetch: (uri: any, options: any) => Promise<Response> = async (
     _uri,
     options
   ) => {
+    if (mockFetchOpts.delay) {
+      if (mockFetchOpts.delay.min > mockFetchOpts.delay.max) {
+        invariant.error(
+          "Please configure a minimum delay that is less than the maximum delay."
+        );
+      } else {
```

Comment:
```suggestion
        invariant(
          mockFetchOpts.delay.min > mockFetchOpts.delay.max,
          "Please configure a minimum delay that is less than the maximum delay."
        );
```

Since `delay` is configured when calling `createSchemaFetch` and not when `mockFetch` is called, I think it would be a bit better DX to 1) `throw` instead of reporting just to the console and 2) `throw` immediately when the delay is not configured properly so that user's tests fail fast (in other words, move this check up to line 41). It might be too easy to miss and/or ignored by just reporting to `console` and give users a false sense that the delay is actually working by allowing existing tests to pass.

Reply:
Bonus incentive here: `invariant.error` would be compiled away in production, so if someone were to run their tests with a prod build, they's never see `invariant.error`.

Reply:
Updated in https://github.com/apollographql/apollo-client/pull/11774/commits/4cd4f388a7745038e245a04b9d5936de2733274d